common/src/main/java/com/emc/mongoose/common/concurrent/DaemonBase.java

```diff
@@ -33,6 +34,9 @@ protected abstract void doClose()
 	public final void start()
 	throws IllegalStateException {
 		if(stateRef.compareAndSet(State.INITIAL, State.STARTED)) {
+			synchronized(state) {
```

Comment:
I got it earlier but then I thought notifyAll has no effects here. And I have made a mistake again.